influxdb3_server/src/http/v1.rs

```diff
@@ -445,6 +454,113 @@ fn convert_ns_epoch(value: Value, precision: Precision) -> Result<Value, anyhow:
     .into())
 }
 
+/// Converts a value from an Arrow `ArrayRef` at a given row index into a `serde_json::Value`.
+///
+/// This function handles various Arrow data types, converting them into their corresponding
+/// JSON representations. For unsupported data types, it returns an error using the `anyhow` crate.
+fn cast_column_value(column: &ArrayRef, row_index: usize) -> Result<Value, anyhow::Error> {
+    let value = match column.data_type() {
+        DataType::Boolean => Value::Bool(column.as_boolean().value(row_index)),
+        DataType::Null => Value::Null,
+        DataType::Int8 => Value::Number(column.as_primitive::<Int8Type>().value(row_index).into()),
+        DataType::Int16 => {
+            Value::Number(column.as_primitive::<Int16Type>().value(row_index).into())
+        }
+        DataType::Int32 => {
+            Value::Number(column.as_primitive::<Int32Type>().value(row_index).into())
+        }
+        DataType::Int64 => {
+            Value::Number(column.as_primitive::<Int64Type>().value(row_index).into())
+        }
+        DataType::UInt8 => {
+            Value::Number(column.as_primitive::<UInt8Type>().value(row_index).into())
+        }
+        DataType::UInt16 => {
+            Value::Number(column.as_primitive::<UInt16Type>().value(row_index).into())
+        }
+        DataType::UInt32 => {
+            Value::Number(column.as_primitive::<UInt32Type>().value(row_index).into())
+        }
+        DataType::UInt64 => {
+            Value::Number(column.as_primitive::<UInt64Type>().value(row_index).into())
+        }
+        DataType::Float16 => Value::Number(
+            serde_json::Number::from_f64(
+                column
+                    .as_primitive::<Float16Type>()
+                    .value(row_index)
+                    .to_f64(),
+            )
+            .context("failed to downcast Float16 column")?,
+        ),
+        DataType::Float32 => Value::Number(
+            serde_json::Number::from_f64(
+                column.as_primitive::<Float32Type>().value(row_index) as f64
```

Comment:
You should be able to use `.into()` instead of `as f64` here.

Reply:
Fixed